tests/end_to_end/end_to_end.cpp

```diff
@@ -29,20 +33,151 @@ namespace ebpf {
+    const in_addr& destination;
+};
+
+struct _ipv6_address_pair
+{
+    const in6_addr& source;
+    const in6_addr& destination;
+};
+
+const in_addr _test_source_ipv4 = {10, 0, 0, 1};
+const in_addr _test_destination_ipv4 = {20, 0, 0, 1};
+const struct _ipv4_address_pair _test_ipv4_addrs = {_test_source_ipv4, _test_destination_ipv4};
+
+const in6_addr _test_source_ipv6 = {0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x2e, 0xfe, 0x12, 0x34};
+const in6_addr _test_destination_ipv6 = {0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x2e, 0xfe, 0x56, 0x78};
+const struct _ipv6_address_pair _test_ipv6_addrs = {_test_source_ipv6, _test_destination_ipv6};
+
+typedef class _udp_packet
+{
+  public:
+    _udp_packet(
+        ADDRESS_FAMILY address_family = AF_INET,
+        _In_ const std::array<uint8_t, 6>& source_mac = _test_source_mac,
+        _In_ const std::array<uint8_t, 6>& destination_mac = _test_destination_mac,
+        _In_opt_ const void* ip_addrs = nullptr,
+        uint16_t datagram_length = 1024)
+        : _address_family(address_family)
+    {
+        UNREFERENCED_PARAMETER(source_mac);
+        UNREFERENCED_PARAMETER(destination_mac);
+        _packet = prepare_udp_packet(
+            sizeof(ebpf::UDP_HEADER) + datagram_length,
+            (address_family == AF_INET) ? ETHERNET_TYPE_IPV4 : ETHERNET_TYPE_IPV6);
+        set_mac_addresses(source_mac, destination_mac);
+        if (_address_family == AF_INET)
+            (ip_addrs == nullptr) ? set_ipv4_addresses(&_test_ipv4_addrs.source, &_test_ipv4_addrs.destination)
+                                  : set_ipv4_addresses(
+                                        &(reinterpret_cast<const _ipv4_address_pair*>(ip_addrs))->source,
+                                        &(reinterpret_cast<const _ipv4_address_pair*>(ip_addrs))->destination);
+        else
+            (ip_addrs == nullptr) ? set_ipv6_addresses(&_test_ipv6_addrs.source, &_test_ipv6_addrs.destination)
+                                  : set_ipv6_addresses(
+                                        &(reinterpret_cast<const _ipv6_address_pair*>(ip_addrs))->source,
+                                        &(reinterpret_cast<const _ipv6_address_pair*>(ip_addrs))->destination);
+    }
+    uint8_t*
+    data()
+    {
+        return _packet.data();
+    }
+    size_t
+    size()
+    {
+        return _packet.size();
+    }
+
+    static const ebpf::UDP_HEADER*
+    _get_udp_header(_In_ const uint8_t* packet_buffer, ADDRESS_FAMILY address_family)
+    {
+        auto eth = reinterpret_cast<const ebpf::ETHERNET_HEADER*>(packet_buffer);
+        const ebpf::UDP_HEADER* udp = nullptr;
+        if (address_family == AF_INET) {
+            auto ip = reinterpret_cast<const ebpf::IPV4_HEADER*>(eth + 1);
+            udp = reinterpret_cast<const ebpf::UDP_HEADER*>(ip + 1);
+        } else {
+            auto ip = reinterpret_cast<const ebpf::IPV6_HEADER*>(eth + 1);
+            udp = reinterpret_cast<const ebpf::UDP_HEADER*>(ip + 1);
```

Comment:
same as above. The fake packets dont have any next header field set. Since this is not going to be used for real packets, I am resolving this as won'f fix.